main.go

```diff
@@ -62,8 +68,7 @@ func main() {
 		log.Info("Server stopped without error")
 	}
 
-	// Perform graceful shutdown
-	asyncServer.InitiateShutdown()
```

Comment:
We can retain this for all the cases and just exit from the signal go routine without doing anything.
Also this enable us to call cancel() just once as part of differ..

Reply:
Please ignore these comments. My bad...